### PR TITLE
Script to migrate data from old DB to new DB schema

### PR DESCRIPTION
**Pull request by anju-james**
Script to extract and transport data from old moc_billing DB to new moc_reporting DB with a different schema.

Test steps:
- Connect to old and Db and new Db
- Connecting to old Db by hardcoding the db name, host, username and password in main method
- Run script: python3 data_migrator.py 
- You should see the test data in new DB schema

### CODE REVIEW

File: data_migrator.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    #institution2poc = etl.fromdb(from_db_conn, 'select * from institution2poc')
+    #inst2poc_transformed = etl.rename(institution2poc, 'poc_role_id', 'role_id')
+    #etl.todb(inst2poc_transformed, to_db_conn, 'institution2poc')
+
+    # Role info added from bill.sql file
+    #role = etl.fromdb(from_db_conn, 'select * from poc_role')
+    #role_transformed = etl.rename(role, {'poc_role_id': 'role_id', 'poc_role_name': 'role_name',
+    #                                     'poc_role_desc': 'role_description',
+    #                                     'poc_role_type': 'role_level'})
+    #etl.todb(role_transformed, to_db_conn, 'role')
```
Comment (kevinliang43) [inline data_migrator.py]: Is this commented-out block of code necessary to keep into the final script?

File: data_migrator.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    item_transformed = etl.cutout(item, 'domain_id')
+    etl.todb(item_transformed, to_db_conn, 'item')
+
+    raw_item_ts_unique = etl.fromdb(from_db_conn, "WITH summary AS ( SELECT its.item_id, its.start_ts, its.end_ts, its.state, its.catalog_item_id, ROW_NUMBER() OVER(PARTITION BY its.item_id) AS rk FROM ITEM_TS its) SELECT s.* FROM summary s WHERE s.rk = 1")
+    raw_item_ts_unique = etl.cutout(raw_item_ts_unique, 'rk')
+    etl.todb(raw_item_ts_unique, to_db_conn, 'raw_item_ts')
+
+
+if __name__ == '__main__':
+    source_db_config = {"dbname":"", "host":"", "user": "", "pass": ""}
```
Comment (tsachleben) [inline data_migrator.py:102]: Put a comment here explaining that this object represents the config/auth to the older Database and needs to be filled in once before running the code.

**Review by tsachleben — CHANGES_REQUESTED** [review]

File: data_migrator.py (hunk 2 above)
Comment (anju-james) [reply data_migrator.py:102]: Added the comment.

File: data_migrator.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    to_db_conn = get_connection(config['host'], config['dbname'], config['user'], config['pass'])
+    return from_db_conn, to_db_conn
+
+def transfer_data(from_db_conn, to_db_conn):
+    '''
+    Transfer data from databases given cursor to execute queries to connected databases
+    Limitations:
+    1. poc.address_id is currently marked as  -1 since it was not provided in test data and is a FK non-null constraint
+    2. institution2poc table is not available in old schema
+    3. role table is already populated in bill.sql file so that table is skipped by this script
```
Comment (rob-baron) [inline data_migrator.py:21]: Should confirm that roles are the same/...

File: data_migrator.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    4. poc_poc_id is currently set to be poc_id since no relevant information is available about the column
+    5. project2moc_project.role_id column is not available in old schema and is a not null field in new schema
+        so we default it to 1 for now.
+    6. project2moc_project.username is not available from old schema so currently set to empty
```
Comment (rob-baron) [inline data_migrator.py:25]: I split the project2poc table into project2poc and project2moc_poc.  Different information has to go into the tables.

**Review by rob-baron — APPROVED** [review]
This is a reasonable start.

**Review by anju-james — COMMENTED** [review]
N.B
- The test data we were provided did not contain the datas or tables for poc_role and project2poc containing any role related information. So we are unable to confirm if the roles the same in old and new data. 
- The different information(username and role_id) to go into  the poc2project and poc2moc_project are currently not available from test data as poc2project table was not provided in test data.

**Review by tsachleben — APPROVED** [review]

**Pull request merged**